### DISCUSSION TITLE
Will laravel "reencrypt" values with new key when rotating?

### DISCUSSION THREAD

**Discussion by noah1400** [laravel/framework#52007] [Q&A]
If I implement key rotation in Laravel, as explained [here](https://laravel.com/docs/11.x/encryption#gracefully-rotating-encryption-keys), will Laravel automatically re-encrypt values with the new key if it can't decrypt them with the new key? If not, how can I implement this behavior?

This would be useful for me because I have a value in my database that is accessed very often. I don't want Laravel to try all keys every time I access this value. Instead, I want Laravel to try the keys once, decrypt the value with the matching key, and then re-encrypt the value with the main key.

**Comment by ghost** [comment]
What you say sounds reasonable and I hope laravel does it like that already.

**Comment by henzeb** [comment]
I wanted to say, it is in the docs, but apparently it is not in the docs... 

The only way to know this is to test it out....
